cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -207,97 +199,116 @@ runQueryTip
   -> ExceptT ShelleyQueryCmdError IO ()
 runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
   SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
-  let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
-      consensusMode = consensusModeOnly cModeParams
 
-  anyEra <- determineEra cModeParams localNodeConnInfo
-  tip <- liftIO $ getLocalChainTip localNodeConnInfo
+  case consensusModeOnly cModeParams of
+    CardanoMode -> do
+      let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
+
+      (chainTip, emLocalState) <- liftIO $ queryQueryTip localNodeConnInfo Nothing
+
+      mLocalState <- fmap join . hushM emLocalState $ \e -> do
+        liftIO . T.hPutStrLn IO.stderr $
+          "Warning: Local state unavailable: " <> renderShelleyQueryCmdError (ShelleyQueryCmdAcquireFailure e)
+
+      let tipSlotNo = case chainTip of
+            ChainTipAtGenesis -> 0
+            ChainTip slotNo _ _ -> slotNo
+
+      mLocalStateOutput :: Maybe O.QueryTipLocalStateOutput <- fmap join . forM mLocalState $ \localState -> do
+        case slotToEpoch tipSlotNo (O.eraHistory localState) of
+          Left e -> do
+            liftIO . T.hPutStrLn IO.stderr $
+              "Warning: Epoch unavailable: " <> renderShelleyQueryCmdError (ShelleyQueryCmdPastHorizon e)
+            return Nothing
+          Right (epochNo, _, _) -> do
+            syncProgressResult <- runExceptT $ do
+              systemStart <- fmap getSystemStart (O.mSystemStart localState) & hoistMaybe ShelleyQueryCmdSystemStartUnavailable
+              nowSeconds <- toRelativeTime (SystemStart systemStart) <$> liftIO getCurrentTime
+              tipTimeResult <- getProgress tipSlotNo (O.eraHistory localState) & bimap ShelleyQueryCmdPastHorizon fst & except
+
+              let tolerance = RelativeTime (secondsToNominalDiffTime 600)
+
+              return $ flip (percentage tolerance) nowSeconds tipTimeResult
+
+            mSyncProgress <- hushM syncProgressResult $ \e -> do
+              liftIO . T.hPutStrLn IO.stderr $ "Warning: Sync progress unavailable: " <> renderShelleyQueryCmdError e
```

Comment:
Same thing here, why the addition `hushM` vs rendering this in the `ToJSON` instance?